kotlinx-coroutines-core/jvm/src/debug/internal/DebugProbesImpl.kt

```diff
@@ -20,24 +20,28 @@ import kotlin.coroutines.jvm.internal.CoroutineStackFrame
 import kotlin.synchronized
 import kotlinx.coroutines.internal.artificialFrame as createArtificialFrame // IDEA bug workaround
 
+/*
+ * This class implement debugger infrastructure is directly used by IDEA debugger.
+ * **DO NOT MAKE BINARY-INCOMPATIBLE CHANGES TO THIS CLASS**.
+ */
+@PublishedApi
 internal object DebugProbesImpl {
     private const val ARTIFICIAL_FRAME_MESSAGE = "Coroutine creation stacktrace"
     private val dateFormat = SimpleDateFormat("yyyy/MM/dd HH:mm:ss")
 
     private var weakRefCleanerThread: Thread? = null
 
     // Values are boolean, so this map does not need to use a weak reference queue
-    private val capturedCoroutinesMap = ConcurrentWeakMap<CoroutineOwner<*>, Boolean>()
-    private val capturedCoroutines: Set<CoroutineOwner<*>> get() = capturedCoroutinesMap.keys
+    private val capturedCoroutinesMap = ConcurrentWeakMap<CoroutineOwnerImpl<*>, Boolean>()
+    private val capturedCoroutines: Set<CoroutineOwnerImpl<*>> get() = capturedCoroutinesMap.keys
 
     @Volatile
     private var installations = 0
 
     /**
-     * This internal method is used by IDEA debugger under the JVM name of
-     * "isInstalled$kotlinx_coroutines_debug".
+     * Internal (JVM-public) getter is used by IDEA debugger.
      */
-    internal val isInstalled: Boolean get() = installations > 0
+    public val isInstalled: Boolean get() = installations > 0
```

Comment:
isn't it going to break https://github.com/JetBrains/kotlin/blob/62dcfcde790e330869ff1fe2ad5dda5e2e9daeb7/idea/jvm-debugger/jvm-debugger-coroutine/src/org/jetbrains/kotlin/idea/debugger/coroutine/proxy/mirror/coroutinesDebugMirror.kt#L22?